vars/spinnakerTriggerPipeline.groovy

```diff
@@ -0,0 +1,176 @@
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixin(parameters, PARAMETER_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .withMandatoryProperty('spinnaker/gateUrl')
+            .withMandatoryProperty('spinnaker/application')
+            .withMandatoryProperty('spinnaker/pipelineNameOrId')
+            .use()
+
+        // telemetry reporting
+        new Utils().pushToSWA([
+            step: STEP_NAME
+        ], config)
+
+        int timeoutMinutes = parameters.timeout ?: 60
+
+        String paramsString = ""
+        if (config.spinnaker.pipelineParameters) {
+            def pipelineParameters = [parameters: config.spinnaker.pipelineParameters]
+
+            paramsString = "-d '${new GStringTemplateEngine().createTemplate(JsonOutput.toJson(pipelineParameters)).make([config: config, env: env]).toString()}'"
+
+            if (config.verbose) {
+                echo "[${STEP_NAME}] Triggering Spinnaker pipeline with parameters: ${paramsString}"
+            }
+        }
+
+        def pipelineTriggerResponse
+
+        //ToDO: support userId/pwd authentication or token authentication!
+
+        def curlVerbosity =  (config.verbose==true) ? '--verbose ' : '--silent '
+
+        withCredentials([
+            file(credentialsId: config.spinnaker.keyFileCredentialsId, variable: 'clientKey'),
+            file(credentialsId: config.spinnaker.certFileCredentialsId, variable: 'clientCertificate')
+        ]) {
+            // Trigger a pipeline execution by calling invokePipelineConfigUsingPOST1 (see https://www.spinnaker.io/reference/api/docs.html)
+            pipelineTriggerResponse = sh(returnStdout: true, script: "curl -H 'Content-Type: application/json' -X POST ${paramsString} ${curlVerbosity}--insecure --cert \$clientCertificate --key \$clientKey ${config.spinnaker.gateUrl}/pipelines/${config.spinnaker.application}/${config.spinnaker.pipelineNameOrId}").trim()
+        }
+        if (config.verbose) {
+            echo "[${STEP_NAME}] Spinnaker pipeline trigger response = ${pipelineTriggerResponse}"
+        }
+
+        def pipelineTriggerResponseObj = readJSON text: pipelineTriggerResponse
+        if (!pipelineTriggerResponseObj.ref) {
+            error "[${STEP_NAME}] Failed to trigger Spinnaker pipeline"
+        }
+
+        if (config.timeout == 0) {
+            echo "[${STEP_NAME}] Exiting without waiting for Spinnaker pipeline result."
+            return
+        }
+
+        echo "[${STEP_NAME}] Spinnaker pipeline ${pipelineTriggerResponseObj.ref} triggered, waiting for the pipeline to finish"
+
+        def pipelineStatusResponseObj
+        def pipelineStatusResponse
```

Comment:
Scope of `pipelineStatusResponse` can be reduced (it is sufficient to have this inside the `waitUtil`

Reply:
done ...